Fix screen share test in SLE 15-SP4

DeepthiYV opened the pull request:
vino is considered out-of-date, and now gnome-remote-desktop progressively provides the screen sharing. 
So I have renamed the  test module `vino_screensharing_available` to `screensharing_available` to verify the availability of screen share across all SLE Versions. If the screen sharing is not available then test does the installation of package `gnome-remote-desktop`  in SLE 15 SP4 and in lower version `vino` package. 

https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/767

- Related ticket:https://progress.opensuse.org/issues/104601
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/-/merge_requests/1597
- Verification run: 
   SLE 15-SP4 : http://10.161.229.147/tests/2055#step/vino_screensharing_available/1
   SLE 15-SP3: https://openqa.suse.de/tests/8323509#step/vino_screensharing_available/1
   SLE 15-SP1: https://openqa.suse.de/tests/8324219#step/vino_screensharing_available/1
   SLE 15-SP2: https://openqa.suse.de/tests/8328802#step/vino_screensharing_available/1
   
   Latest Verification run:
  [SLE 15-SP4](http://10.161.229.147/tests/2075#step/screensharing_available/1) | [SLE 15-SP3](https://openqa.suse.de/tests/8361120#step/screensharing_available/1) | [SLE 15-SP2](https://openqa.suse.de/tests/8361800#step/screensharing_available/1) | [SLE 15-SP1](https://openqa.suse.de/tests/8362903#step/screensharing_available/1) | [LEAP 15 aarch64](https://openqa.opensuse.org/tests/2256530#step/screensharing_available/1) | [LEAP 15 PPC64](https://openqa.opensuse.org/tests/2256529#step/screensharing_available/1)

dzedro commented on tests/x11/remote_desktop/vino_screensharing_available.pm:
```diff
@@ -35,18 +34,24 @@ sub run {
-
-        # Install the vino package which is probably the case of missing screen sharing option
-        ensure_installed 'vino';
+        if (is_sle("=15-sp4")) {
```
I would use 15-SP4+, I don't think on next SLE version it will go back to vino

DeepthiYV replied:
Thanks, I have updated the code to 15-SP4+.

DeepthiYV commented:
> Hi Dee, could you explain a bit more the purpose of the PR? In the description you write `Replaced vino to gnome-remote-desktop`, but the module is called `vino_screensharing_available`
Anna,  Please let me above description is OK?

DeepthiYV commented:
> > Hi Dee, could you explain a bit more the purpose of the PR? In the description you write `Replaced vino to gnome-remote-desktop`, but the module is called `vino_screensharing_available`
> > Anna,  Please let me know if above description is OK?

DeepthiYV commented:
> Hi Dee, could you explain a bit more the purpose of the PR? In the description you write `Replaced vino to gnome-remote-desktop`, but the module is called `vino_screensharing_available`

Hi Anna, I updated the description above and let me know is this ok :)

foursixnine commented on tests/x11/remote_desktop/vino_screensharing_available.pm:
```diff
@@ -35,18 +34,24 @@ sub run {
+        if (is_sle("=15-sp4")) {
+            record_info 'gnome-remote-desktop', 'After the installation the screen sharing is not available - gnome-remote-dekstop is missing and we need to install it now.';
+            send_key 'ctrl-q';
+            # Install the vino package which is probably the case of missing screen sharing option
```
```suggestion
            # Install the gnome-desktop package which is probably the case of missing screen sharing option
```

punkioudi commented:
> > Hi Dee, could you explain a bit more the purpose of the PR? In the description you write `Replaced vino to gnome-remote-desktop`, but the module is called `vino_screensharing_available`
> 
> Hi Anna, I updated the description above and let me know is this ok :)

Still the title of the module seems misleading to me :P

punkioudi commented on tests/x11/remote_desktop/vino_screensharing_available.pm:
```diff
@@ -35,18 +34,24 @@ sub run {
+            ensure_installed 'gnome-remote-desktop';
+        }
+        else {
+            record_info 'vino missing', 'After the installation the screen sharing is not available - vino is missing and we need to install it now.';
```
Replace also vino here with gnome-desktop?

foursixnine commented on tests/x11/remote_desktop/vino_screensharing_available.pm:
```diff
@@ -35,18 +34,24 @@ sub run {
+            record_info 'vino missing', 'After the installation the screen sharing is not available - vino is missing and we need to install it now.';
+            send_key 'ctrl-q';
 
+            # Install the vino package which is probably the case of missing screen sharing option
+            ensure_installed 'vino';
```
Please add a comment on why vino is still tested, i.e (Or a link where this has been mentioned or discusse)

foursixnine commented on tests/x11/remote_desktop/vino_screensharing_available.pm:
```diff
@@ -64,7 +69,7 @@ sub run {
     } else {
         assert_screen 'with_screensharing';
-        record_info 'vino present', 'Vino and the screen sharing are present';
+        is_sle("=15-sp4") ? record_info('gnome-remote-desktop present and the screen sharing are present') : record_info('vino present', 'Vino and the screen sharing are present');
```
see above comment from @dzedro

DeepthiYV replied:
Updated the comment. Thanks :)

DeepthiYV commented:
> > Hi Dee, could you explain a bit more the purpose of the PR? In the description you write `Replaced vino to gnome-remote-desktop`, but the module is called `vino_screensharing_available`
> 
> Hi Anna, I updated the description above and let me know is this ok :)

Sure Anna,  As we discussed offline, I have renamed module to generic name and verified the test. Please let me know your feedback. Thanks :)

dzedro commented on lib/main_common.pm:
```diff
@@ -1309,6 +1309,7 @@ sub load_x11tests {
                 loadtest "x11/eog";
                 loadtest(is_sle('<15') ? "x11/rhythmbox" : "x11/gnome_music");
                 loadtest "x11/wireshark";
+                loadtest "x11/remote_desktop/screensharing_available" if is_sle("=15-sp4");
```
Why ? If difference between 15sp4+ and older versions is handled in the test itself.

foursixnine commented on tests/x11/remote_desktop/screensharing_available.pm:
```diff
@@ -35,18 +34,23 @@ sub run {
-        ensure_installed 'vino';
-
+        if (is_sle("15-sp4+")) {
+            record_info 'gnome-remote-desktop', 'After the installation the screen sharing is not available - gnome-remote-dekstop is missing and we need to install it now.';
```
is there a bug for this? I think this should be a soft failure (same below)... I.E we should know what's the expectation and expected result after the installation?. Maybe a separate ticket to figure that out and update later (and actually use the screen sharing)... to keep things moving

DeepthiYV replied:
In SLE15 SP4, gnome-remote-desktop is available by default and so screen sharing, I added check similar to earlier condition. But it is gnome-remote-desktop instead of vino for SP4.

foursixnine commented on lib/main_common.pm:
```diff
@@ -1335,6 +1335,7 @@ sub load_x11tests {
                 loadtest(is_sle('<15') ? "x11/rhythmbox" : "x11/gnome_music");
                 loadtest "x11/wireshark";
                 loadtest "x11/ImageMagick";
+                loadtest "x11/remote_desktop/screensharing_available" if is_sle("15-sp4+");
```
Once you create the ticket for the extraction, please add the link to this PR

foursixnine replied:
https://progress.opensuse.org/issues/109373

foursixnine commented:
Jozef is still FTO
